tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -910,25 +900,26 @@ public ImmutableTracerSettings Build()
             return new ImmutableTracerSettings(this);
         }
 
-        internal void CollectTelemetry(IConfigurationTelemetry destination)
+        internal static IDictionary<string, string>? InitializeServiceNameMappings(ConfigurationBuilder config, string key)
         {
-            // copy the current settings into telemetry
-            _telemetry.CopyTo(destination);
+            return config
+               .WithKeys(key)
+               .AsDictionary()
+              ?.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
+               .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
+        }
 
-            // record changes made in code directly to destination
-            _initialSettings.RecordChanges(this, destination);
+        internal static IDictionary<string, string>? InitializeHeaderTags(ConfigurationBuilder config, string key, bool headerTagsNormalizationFixEnabled)
```

Comment:
That's because I distinguish "empty" from "not set". It makes no difference for the settings set through environment variables, but it does for dynamic config since they're going to override the configuration only if they're set.